vars/piperExecuteBin.groovy

```diff
@@ -25,6 +26,23 @@ void call(Map parameters = [:], stepName, metadataFile, List credentialInfo, fai
 
         writeFile(file: ".pipeline/tmp/${metadataFile}", text: libraryResource(metadataFile))
 
+        // The default pipeline config, plus any custom default configs, have been
+        // extracted from the library resources into .pipeline/ by setupCommonPipelineEnvironment.groovy
+        // and fed into the DefaultValueCache. The 'default_pipeline_environment.yml' file
+        // is not reflected in the list returned by getCustomDefaults(), but has to be
+        // included in the --defaultConfig param. This makes sure that the same default configuration
+        // values are visible from the go side as from the Groovy side.
+        List customDefaults = ['default_pipeline_environment.yml']
+        customDefaults.addAll(DefaultValueCache.getInstance().getCustomDefaults())
+        for (int i = 0; i < customDefaults.size(); i++) {
+            customDefaults[i] = '".pipeline/' + customDefaults[i] + '"'
```

Comment:
Maybe with applying bash utils escape: 
```suggestion
            customDefaults[i] = ".pipeline/${customDefaults[i]}"
```

Reply:
Can we be sure that it was copied into `.pipeline`. Should we maybe copy it always as part of this step? 

@OliverNocon Why have we copied it at all before, when it was not used?

Reply:
I've used `BashUtils` now (tested locally that the arguments still work as "list of strings" if they end up as `'defaultFile1','defaultFile2'`).

Reply:
> @OliverNocon Why have we copied it at all before, when it was not used?

This is a good question.

> Can we be sure that it was copied into `.pipeline`. Should we maybe copy it always as part of this step?

We could extract what currently happens in `setupCommonPipelineEnvironment.groovy` into its own method, and check with `fileExists()` for each custom config. Maybe move this code into the `DefaultValueCache` as a static method? Then it can be called from `piperExecuteBin` as well.